thing-url-adapter.js

```diff
@@ -798,10 +839,34 @@ function loadThingURLAdapter(addonManager) {
     if (typeof config.pollInterval === 'number') {
       adapter.pollInterval = config.pollInterval * 1000;
     }
-
     for (const url of config.urls) {
-      adapter.loadThing(url);
+      if ('authentication' in url) {
```

Comment:
Ok, the biggest thing left to do here is to transition from the old config format (i.e. just an array of strings) to the new format (i.e. an array of objects with the `href` property set). There's a good example of that process [here](https://github.com/WebThingsIO/serial-adapter/blob/master/serial-adapter.js#L336-L359).

Reply:
I was wondering about this. 

So first thing when the db opens - check the format and if needed convert it to the new format and save, then continue on.

Reply:
Yes, exactly.

Reply:
I'm noticing on the adapter config page that an authentication method is required. Shouldn't this be optional? If I just convert a string url  to object with href set and no auth info the config page won't load.

Reply:
The entire authentication object is optional, with the method only being required if the authentication object is present. When you're updating the config, are you completely omitting the authentication object, or creating it with an empty object?

Reply:
I'm doing it like this: 

 let modified = false;
    for (const entry in config.urls) {
      if (typeof config.urls[entry] === 'string') {
        config.urls[entry] = {href: config.urls[entry]};
        modified = true;
      }
    }
    if (modified) {
      db.saveConfig(config);
    }


When I try to save an entry with no method set it tells me it is required. 

![Screen Shot 2020-09-30 at 12 32 51 PM](https://user-images.githubusercontent.com/2983414/94731090-149f9300-0319-11eb-8ca0-a86f5ad21f89.png)


Maybe I copied the manifest part wrong -- here it is https://github.com/m2ag-labs/thing-url-adapter/blob/master/manifest.json

Reply:
That's probably a UI bug. You could add a "none" option to the method enum and set the old config values to use that.

Reply:
Done

Reply:
Done